### PR TITLE
Part 5: Controllers specs

### PR DESCRIPTION
**Pull request by dfsalam**
# In this pull request I:

## Fixed linter errors.

### CODE REVIEW

File: spec/requests/users_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+RSpec.describe 'UsersController', type: :request do
+  describe 'GET /index' do
+    it 'Returns http success' do
+      get '/users'
+      expect(response.status).to eq(200)
+    end
+    it 'renders the index template' do
+      get '/users'
+      expect(response).to render_template(:index)
+    end
```
Comment (omar-muhamad) [inline spec/requests/users_controller_spec.rb:12]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=template%20was%20rendered.-,If%20the%20response%20body%20includes%20correct%20placeholder%20text.,-Need%20a%20big), Please include a test that ensures the response body includes the correct placeholder text.
Example : `expect(response.body).to include("Welcome to user index page")`

File: spec/requests/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      get "/users/#{first_user_id}"
+      expect(response.status).to eq(200)
+    end
+    it 'renders the show template' do
+      get "/users/#{first_user_id}"
+      expect(response).to render_template(:show)
+    end
+    it 'Returns not found' do
+      expect { get '/users/789' }.to raise_error(ActiveRecord::RecordNotFound)
+    end
```
Comment (omar-muhamad) [inline spec/requests/users_controller_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=template%20was%20rendered.-,If%20the%20response%20body%20includes%20correct%20placeholder%20text.,-Need%20a%20big), Kindly replace this test with another one that ensures the response body includes the correct placeholder text.
Example : expect(response.body).to include("Welcome to user show page")

File: spec/requests/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      get "/users/#{second_user_id}/posts"
+      expect(response.status).to eq(200)
+    end
+    it 'renders the index template' do
+      get "/users/#{second_user_id}/posts"
+      expect(response).to render_template(:index)
+    end
+    it 'Returns not found' do
+      expect { get '/users/789/posts' }.to raise_error(ActiveRecord::RecordNotFound)
+    end
```
Comment (omar-muhamad) [inline spec/requests/posts_controller_spec.rb]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=template%20was%20rendered.-,If%20the%20response%20body%20includes%20correct%20placeholder%20text.,-Need%20a%20big), Kindly replace this test with another one that ensures the response body includes the correct placeholder text.
Example : expect(response.body).to include("Welcome to post index page")

File: spec/requests/posts_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    it 'renders the show template' do
+      get "/users/#{second_user_id}/posts/#{first_post_id}"
+      expect(response).to render_template(:show)
+    end
+    it 'Returns not found' do
+      expect { get "/users/789/posts/#{first_post_id}" }.to raise_error(ActiveRecord::RecordNotFound)
+    end
+    it 'Returns not found' do
+      expect { get "/users/#{second_user_id}/posts/789" }.to raise_error(ActiveRecord::RecordNotFound)
+    end
```
Comment (omar-muhamad) [inline spec/requests/posts_controller_spec.rb]: - [ ] Same here, As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=template%20was%20rendered.-,If%20the%20response%20body%20includes%20correct%20placeholder%20text.,-Need%20a%20big), Kindly replace those tests with one test that ensures the response body includes the correct placeholder text.
Example : expect(response.body).to include("Welcome to post show page")

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @dfsalam,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @dfsalam, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**